ap/tasks/framework/mixins.py

```diff
@@ -127,19 +161,15 @@ class SelectorMixin(ConfigTask):
         "'default_selector' config",
     )
 
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-
-        # cache for selector func
-        self._selector_func = None
+    update_selector = False
 
     @classmethod
     def modify_param_values(cls, params):
         params = super().modify_param_values(params)
 
         # add the default selector when empty
-        if "config" in params and params.get("selector") == law.NO_STR:
-            config_inst = cls.get_analysis_inst(cls.analysis).get_config(params["config"])
+        if "config_inst" in params and params.get("selector") == law.NO_STR:
+            config_inst = config_inst = params["config_inst"]
```

Comment:
There is a "typo" here (I guess it doesn't make a big difference :D )
```suggestion
             config_inst = params["config_inst"]
```
This is exactly the same as in the `CalibratorMixin` class, right? Does it make sense to centralize this logic in the base class for both/ all `Mixin` classes?